### PR TITLE
fix weather.py to work with new Yahoo API changes

### LINKED ISSUE #1029: .setlocation gives an error and fails: AttributeError: 'NoneType' object has no attribute 'get'
yoosi: `.setlocation london`

sopel: `AttributeError: 'NoneType' object has no attribute 'get' (file "/home/sopel/sopel/sopel/modules/weather.py", line 181, in update_woeid)`

### LINKED ISSUE #1050: weather now requiring oauth
The weather module seems to be requiring oauth now.
Manually running through the command would yield this: "Please provide valid credentials. OAuth oauth_problem='OST_OAUTH_PARAMETER_ABSENT_ERROR', realm='yahooapis.com'"

From yahoo's dev portal: [https://developer.yahoo.com/weather/](https://developer.yahoo.com/weather/)
"To increase the security measure and better serving our YDN customers, weather data source will enforce OAuth 1 starting March 15, 2016. Please update your API calls with OAuth 1 enabled (via the OAuth /v1/yql end point if using YQL) using your Yahoo App key and secret."

### PR DESCRIPTION
**Pull request by yoosi**
this closes #1050 and closes #1029

Yahoo no longer provides RSS feeds for weather.
bot now uses their most current API instead.

### CODE REVIEW

**Comment by maxpowa** [comment]
This is good but it is still using XML parsing - with the `query.yahooapis.com` urls, we can use `&format=json` to avoid XML completely.

**Comment by yoosi** [comment]
@maxpowa I'm still a bit new to git. Should a change like that get its own pull request or should I commit to this branch?

**Comment by maxpowa** [comment]
New PR. As it is, this one doesn't fix the problem - I tested it the other day and it was still erroring.

**Comment by yoosi** [comment]
@maxpowa What errors are you receiving? I've been running this updated module since I opened my pull request without issue.

File: sopel/modules/weather.py [hunk 1]
```diff
@@ -178,6 +178,7 @@ def update_woeid(bot, trigger):
         city = first_result.get('name')
     state = first_result.get('admin1').get('#text') or ''
     country = first_result.get('country').get('#text') or ''
-    uzip = first_result.get('postal').get('#text') or ''
-    bot.reply('I now have you at WOEID %s (%s%s, %s, %s %s)' %
-              (woeid, neighborhood, city, state, country, uzip))
+    # uzip = first_result.get('postal').get('#text') or ''
```
Comment (dgw) [inline sopel/modules/weather.py]: This whole section, from here to the end of the diff, is messy. The commented-out code should simply be removed.

**Comment by yoosi** [comment]
@dgw I removed the messy code

**Comment by dgw** [comment]
:+1: though I suspect that these commits should all be squashed into one before merging.

File: sopel/modules/weather.py [hunk 2]
```diff
@@ -178,6 +178,5 @@ def update_woeid(bot, trigger):
         city = first_result.get('name')
     state = first_result.get('admin1').get('#text') or ''
     country = first_result.get('country').get('#text') or ''
-    uzip = first_result.get('postal').get('#text') or ''
```
Comment (yoosi) [inline sopel/modules/weather.py]: Yahoo's new API doesn't always return a zip code

File: sopel/modules/weather.py [hunk 3]
```diff
@@ -137,16 +137,16 @@ def weather(bot, trigger):
     if not woeid:
         return bot.reply("I don't know where that is.")
 
-    query = web.urlencode({'w': woeid, 'u': 'c'})
-    raw = web.get('http://weather.yahooapis.com/forecastrss?' + query,
+    query = 'q=select * from weather.forecast where woeid="%s" and u=\'c\'' % woeid
+    body = web.get('http://query.yahooapis.com/v1/public/yql?' + query,
```
Comment (yoosi) [inline sopel/modules/weather.py]: still using deprecated web.get. move past that on another pull request later

File: sopel/modules/weather.py [hunk 4]
```diff
@@ -137,16 +137,16 @@ def weather(bot, trigger):
...
                   dont_decode=True)
-    parsed = xmltodict.parse(raw).get('rss')
-    location = parsed.get('channel').get('title')
-
-    cover = get_cover(parsed)
-    temp = get_temp(parsed)
-    humidity = get_humidity(parsed)
-    wind = get_wind(parsed)
+    parsed = xmltodict.parse(body).get('query')
```
Comment (yoosi) [inline sopel/modules/weather.py]: new API provides JSON so future pull request will move past XML

**Comment by yoosi** [comment]
@dgw I squashed the commits. It's much cleaner now.

**Comment by dgw** [comment]
:+1: @yoosi

**Pull request merged**